include/hipSYCL/sycl/device.hpp

```diff
@@ -312,7 +312,23 @@ HIPSYCL_SPECIALIZE_GET_INFO(device, max_compute_units)
 HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_dimensions)
 { return 3; }
 
-HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_sizes)
+HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_sizes<1>)
+{
+  std::size_t size0 = static_cast<std::size_t>(get_rt_device()->get_property(
+      rt::device_uint_property::max_group_size0));
+  return id<1>{size0};
+}
+
+HIPSYCL_SPECIALIZE_GET_INFO(device, max_work_item_sizes<2>)
+{
+  std::size_t size0 = static_cast<std::size_t>(get_rt_device()->get_property(
+      rt::device_uint_property::max_group_size0));
+  std::size_t size1 = static_cast<std::size_t>(get_rt_device()->get_property(
+      rt::device_uint_property::max_group_size1));
+  return id<2>{size0, size1};
```

Comment:
Here we need to be extra careful. The main motivation for switching to structs instead of the old enums in SYCL 2020 was exactly this query: SYCL implementations for some cases need to flip the dimensions for some backends.
 E.g. on CUDA, the "vectorized" index is always x (i.e. 0), while the SYCL memory layout wants the highest index to be the fast one.
So, on such backends, SYCL implementations flip indices such that from the perspective of the hardware, x is the fast index, while from the application point of view it remains the highest index.

I believe that these runtime queries return the directly the result from the backend for the respective dimension (Can you double check?). Therefore, to make everything match, we would need to check here whether a flip is required (this might require a new property `needs_index_flip` or similar for the runtime), and if so, return `{size1, size0}`.

Reply:
Yeah, I think you suggested that in #832, I'll look into that...

Reply:
It's probably true that currently we might be able to get by by just flipping in any case. HIP, CUDA and Level Zero does the flip, and OpenMP does not (but exposes similar limits in all dimensions as you point out - although it's questionable whether this is actually correct).

My concern is that this introduces an assumption that might not hold anymore in the future e.g. if new backends are introduced, and might be surprising then. To me it seems the better strategy long term to ask those components that actually know the details (such as the runtime backends) instead of making assumptions in the SYCL headers.

Is there a problem introducing new runtime queries?

An alternative to asking whether things should flip could be to directly add a runtime query for the 1, 2, and 3-dimensional maximums. That might be even more flexible.

Reply:
No, there is no problem, it's just I guess I started wondering if it was worth the trouble.
Please have a look now, do I just leave the properties hardcoded or is there a better way of querying the backends?

Reply:
It looks good to me. What do you mean by hardcoded properties for querying the backends?

Reply:
So, you knew that...

> HIP, CUDA and Level Zero does the flip, and OpenMP does not 

But I didn't. So I was wondering if there is _a_ property of the respective runtimes I could query that would expose the need for flipping instead of just returning the boolean literals explicitly like we're doing now.